src/tool.rs

```diff
@@ -16,34 +17,27 @@ fn find_tool() -> Result<(PathBuf, ToolFamily)> {
         };
     }
 
-    // Check with priorities, and retrieve the full compiler path and the
-    // ToolFamily
-    //  * first: clang,
-    //  * second: Windows -> clang-cl _ -> Gnu
-    //  * third Windows -> msvc
+    // TODO : On windows try to put mscv on the environment first
     if exists_command!("clang") {
-        Ok((which::which("clang").unwrap(), ToolFamily::Clang))
-    } else if cfg!(target_os = "windows") {
-        if exists_command!("clang-cl") {
-            Ok((
-                which::which("clang-cl").unwrap(),
-                ToolFamily::Msvc { clang_cl: true },
-            ))
-        } else if exists_command!("cl") {
-            Ok((
-                which::which("cl").unwrap(),
-                ToolFamily::Msvc { clang_cl: false },
-            ))
-        } else if exists_command!("gcc") {
-            Ok((which::which("gcc").unwrap(), ToolFamily::Gnu))
-        } else {
-            Err(Error::NoCompilerFound)
-        }
-    } else if exists_command!("gcc") {
-        Ok((which::which("gcc").unwrap(), ToolFamily::Gnu))
-    } else {
-        Err(Error::NoCompilerFound)
+        return Ok((which::which("clang").unwrap(), ToolFamily::Clang));
+    }
+    if exists_command!("clang-cl") {
```

Comment:
I had it the other way because the msvc variants are unavaible on linux and macos, so searching for them is useless, I'd like clang to always be the first tier compiler on all platforms, as it has some diffs with gcc sometimes, this way if your code has crossplatform in mind and you are using the recomended toolset it won't cause issues with compiler related things.

first tier(all): clang
second tier(windows): clang-cl
second tier(linux): gcc
third tier(windows): msvc

Reply:
I understand